test/libsolidity/Metadata.cpp

```diff
@@ -334,6 +334,119 @@ BOOST_AUTO_TEST_CASE(metadata_revert_strings)
 	BOOST_CHECK_EQUAL(metadata["settings"]["debug"]["revertStrings"], "strip");
 }
 
+BOOST_AUTO_TEST_CASE(metadata_license_missing)
+{
+	CompilerStack compilerStack;
+	char const* sourceCode = R"(
+		pragma solidity >=0.0;
+		contract C {
+		}
+	)";
+	compilerStack.setSources({{"", std::string(sourceCode)}});
+	BOOST_REQUIRE_MESSAGE(compilerStack.compile(), "Compiling contract failed");
+
+	std::string const& serialisedMetadata = compilerStack.metadata("C");
+	BOOST_CHECK(solidity::test::isValidMetadata(serialisedMetadata));
+	Json::Value metadata;
+	BOOST_REQUIRE(util::jsonParseStrict(serialisedMetadata, metadata));
+
+	BOOST_CHECK_EQUAL(metadata["sources"].size(), 1);
+	BOOST_CHECK(metadata["sources"].isMember(""));
+	BOOST_CHECK(!metadata["sources"][""].isMember("license"));
+}
+
+BOOST_AUTO_TEST_CASE(metadata_license_gpl3)
+{
+	CompilerStack compilerStack;
+	char const* sourceCode = R"(
+		// NOTE: we also add trailing whitespace after the license, to see it is trimmed.
+		// SPDX-License-Identifier: GPL-3.0    
+		pragma solidity >=0.0;
+		contract C {
+		}
+	)";
```

Comment:
You could silence @stackenbotten by ditching the raw string in this particular test:
```suggestion
	char const* sourceCode =
		"// NOTE: we also add trailing whitespace after the license, to see it is trimmed."
		"// SPDX-License-Identifier: GPL-3.0    "
		"\"pragma solidity >=0.0;\""
		"contract C {"
		"}";
```

Reply:
Good idea!